### DISCUSSION TITLE
Hello, how include new class in app?

### DISCUSSION THREAD

**Discussion by Nikita2552** [hydrogen-music/hydrogen#1578] [Hydrogen on Linux]
hello, i want doing new feature for hydrogen. I created new button and new class, how i can include it in application, what i must do with cmake and make? I am inexperienced, help  me, please. I really want to help with the project. I have thoughts on what to add.

**Comment by theGreatWhiteShark** [comment]
Hey @Nikita2552 ,

There have been some major changes in the widgets between 1.1 and the upcoming 1.2 release. Now, the button widget is defined in https://github.com/hydrogen-music/hydrogen/blob/master/src/gui/src/Widgets/Button.h and it is only required to create something new in case it features an icon instead of text.

But in general it's best to open a discussion about the feature you have in mind in here first.
